src/protocol/sort/mod.rs

```diff
@@ -4,22 +4,63 @@ use std::fmt::Debug;
 mod apply;
 pub mod bit_permutations;
 mod compose;
+mod generate_sort_permutations;
 pub mod reshare;
 mod secureapplyinv;
 mod shuffle;
 
 #[allow(clippy::module_name_repetitions)]
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
 pub enum SortStep {
-    BitPermutations,
+    ModulusConversion(u8),
+    BitPermutation(u8),
+    ApplyInv(u8),
+    ComposeStep(u8),
```

Comment:
I think this is great! It is simultaneously clear that we are doing the same type of step over and over for each bit, but we can generate a unique step ID. A u8 should be sufficient as we will never need to sort 256-bit numbers =).

Reply:
Might I suggest that you construct something more like:

SortBit(u8) : Step >> enum { ModulusConversion, BitPermutation, ApplyInv, Compose }

!  That is, nest this so that you get "bit45/modconv" or "bit7/bitperm".  Then we only have to do the `AsRef<str>` implementation that is indexed by a u8 once.

Reply:
Just realized that modconv would run per record (vs all other protocols running per bit) so its size would be usize. How should we be handling that?

Reply:
Don't you just convert each bit as a value in Fp2 into a value in F?  And each record is indexed with a RecordId.

Reply:
Yes thats what we need. However, we have been needing to narrow the context for record else step complains of duplicate context. Example here of narrowing:  https://github.com/private-attribution/ipa/blob/main/src/protocol/modulus_conversion/convert_shares.rs#L245
@martinthomson @benjaminsavage